docs/async/async-csharp-vb.rst

```diff
@@ -0,0 +1,213 @@
+
+	    NetworkProgressBar.IsEnabled = false
+	    NetworkProgressBar.Visbility = Visibility.Collapsed
+	End Sub
+
+Web Service snippet (ASP.NET MVC):
+
+.. code-block:: vb.net
+
+	Private Readonly Dim _httpClient As HttpClient = new HttpClient()
+
+	<HttpGet>
+	<Route("MicrosoftCount")>
+	Public Async Function GetDotNetCountAsync() As Task(Of String)
+
+	    ' Suspends GetDotNetCountAsync() to allow the caller (the web server) to accept another request,
+	    ' rather than blocking on this one.
+	    Dim html As String = Await _httpClient.GetStringAsync("https://dotnetfoundation.org");
+
+	    Return Regex.Matches(html, ".NET").Count
+	End Function
+
+More on Task and Task<T>
+------------------------
+
+As mentioned before, Tasks are constructs used to represent operations which you will need to wait on to complete.
+
+* ``Task`` represents a single operation which does not return a value.
+* ``Task<T>`` represents a single operation which returns a value of type ``T``.
+
+It's important to reason about Tasks of abstractions of work happening in the background, and *not* dedicated threads on the thread pool, because in the context of I/O-bound code, they aren't new threads at all!
+
+Here's a 10,000 foot view of what happens with a typical async call:
+
+The call (such as ``GetStringAsync`` from ``HttpClient``) makes its way through the .NET libraries until it reaches a system interop call (such as ``P/Invoke`` on Windows).  This eventually makes the proper System API call (such as ``write`` to a socket file descriptor on Linux).  That System API call is then dealt with in the kernel, where the I/O request is sent to the proper subsystem.  Although details about scheduling the work on the appropriate device driver are different for each OS, eventually an "incomplete task" signal will be sent from the device driver, bubbling its way back up to the .NET runtime.  This will be converted into a ``Task`` or ``Task<T>`` by the runtime, sending that handle to the executing method which invoked the task in the first place.
+
+Although many details were glossed over (such as how "borrowing" compute time on a thread pool is coordinated), the important thing to recognize here is that **no thread is 100% dedicated to running the initiated task**.  This allows threads in the thread pool of a system to handle a larger volume of work rather than having to wait for I/O to finish.
+
+Tasks are awaitable, meaning that the use ``await`` will allow your application or service to perform useful work while the task is running by yielding control to its caller until the task is done.  If you're using ``Task<T>``, the ``await`` keyword will additionally "unwrap" the value returned when the Task is complete.
+
+Although the above may seem like a lot of work to be done, when measured in terms of execution time, it's miniscule compared to the time it takes to do the actual I/O work.  Although not at all precise, a potential timeline for such a call would look like this:
```

Comment:
We usually like to use two precise terms: CPU time, wall-clock time. They are clear, whereas "execution time" can refer to either.